### ISSUE TITLE
Prepare rancher-vsphere-csi chart for 2.7.0

### ISSUE THREAD

**Issue by MKlimuszka** [rancher/rancher#38963]
rancher-vsphere-csi chart needs to be updated the charts for 2.7 by

- File a PR removing the doNotRelease flag form the package.yaml; in this PR, you will also want to make any changes that need to be made to update the chart to the version you intend to release in 2.7. For example, for a charts like rancher-aks-operator / rancher-gke-operator / rancher-eks-operator, you will need to create a new branch in upstream for Rancher 2.7, cut the first release for that chart, and introduce those changes in to the package.yaml as well before removing the doNotRelease flag

- To make your chart available for 2.7, almost every chart will also have to modify the Rancher Version annotations that scope it to 2.6, i.e. https://github.com/rancher/charts/blob/2e058e93ee01f2980e4750148842dd7f04262b38/charts/fleet-agent/100.0.5%2Bup0.3.11/Chart.yaml#L8

- Once the PR is merged, ensure that the brand new chart is smoke tested by QA/dev

**Comment by sowmyav27** [comment]
To validate
- Deploy 2.7.0-rc1
- Deploy vsphere cluster and enable CSI/CPI
- Verify chart versions are
```
rancher-vsphere-cpi:
- 101.0.0+up1.2.2
rancher-vsphere-csi:
- 101.0.0+up2.5.1-rancher1
```

**Comment by vivek-infracloud** [comment]
Validated that the expected charts are available on rancher v2.7.0-rc3

Validation steps

1. Deployed a rancher v2.7.0-rc3
2. Provisioned a k8s cluster `v1.23.12-rancher1-1` with cloud provider selected as `external`
3. Made sure the CPI (v101.0.0+up1.2.2) and CSI (v101.0.0+up2.5.1-rancher1) charts can be installed.
4. Deployed a workload with volume attached to it.

Both CPI and CSI charts are working as expected. Hence closing the issue.
